### DISCUSSION TITLE
compiled binaries for Lambda arm

### DISCUSSION THREAD

**Discussion by mdavis-xyz** [pydantic/pydantic#4699] [Pydantic V2]
I currently deploy pydantic into Amazon lambda functions. Some of my lambdas run in x86, others in arm64 (Graviton).

The [blog post](https://pydantic-docs.helpmanual.io/blog/pydantic-v2/) says new pydantic will support 

> Linux: x86_64, aarch64, i686, armv7l, musl-x86_64 & musl-aarch64

Is `armv7l` compatible with `arm64`? [This post suggests not](https://askubuntu.com/questions/928227/is-armv7l-32-or-64-bit/1042272#1042272). 

(The [Amazon docs](https://aws.amazon.com/blogs/compute/migrating-aws-lambda-functions-to-arm-based-aws-graviton2-processors/) say:

> The Graviton2 processor uses the Neoverse-N1 core and supports Arm V8.2 (include CRC and crypto extensions) plus several other architectural extensions.

Will the new pydantic library be deployable on arm64 in Lambda? If not, that's probably a deal-breaker for me.

**Comment by samuelcolvin** [comment]
Yes it will be, see https://pypi.org/project/pydantic-core/#files

**Comment by mdavis-xyz** [comment]
The only mention of ARM I see there is ARM for Mac, and armv7l.

Will the Mac one (presumably M1?) work on Amazon Linux Graviton?

**Reply by samuelcolvin** [reply]
Why don't you try installing pydantic-core and find out?

**Comment by gaspardbalthazar** [comment]
Did you find a solution?
Using my lambda build with pip install with --platform manylinux2014_aarch64 --python 3.9 --only-binary=:all:
pydantic cpython arrch64 files are not present. Is it a problem to run the lambda with graviton?

**Reply by mdavis-xyz** [reply]
Not sure. I didn't get around to testing pydantic-core on Lambda with graviton.

**Reply by samuelcolvin** [reply]
I believe it should work, best to try it on the hardware and see what happens.

**Reply by dmontagu** [reply]
@davidhewitt pointed this out:

> At https://pypi.org/project/pydantic-core/#files I see [pydantic_core-0.23.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl](https://files.pythonhosted.org/packages/79/ed/21d84b8423205bc4ae367dd6673227480e0915113c59d3311c2b5b07429e/pydantic_core-0.23.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl)

Is this not the file you need @gaspardbalthazar?

**Comment by gaspardbalthazar** [comment]
Thank you it is indeed this file but for pydantic v1.10.7 because in my project pydantic comes from a dependency with fastapi.
When I search https://pypi.org/project/pydantic/#files there is macosx, win but not smanylinux in arm64, do you know why?

**Reply by samuelcolvin** [reply]
V1 you can just install with `no_binary`, v1 works fine as a pure python package.

**Comment by yarmoliq** [comment]
I have encountered this issue as well while trying to deploy telegram bot to graviton aws lambda. This is how I fixed the issue with pydantic (`pydantic` is a dependency for `aiogram`):
```bash
pip install aiogram -t package/ --only-binary=:all: --platform manylinux2014_aarch64
```

This is where i found the platform name for pydantic (graviton aws lambda is `ARM64`): https://pypi.org/project/pydantic-core/#files
